src/Symfony/Component/VarDumper/Dump.php

```diff
@@ -0,0 +1,160 @@
+ * @method self withMaxStringLength(int $length)                   Limits the number of characters for dumped strings.
+ * @method self withMaxStringWidth(int $width)                     Limits the number of characters per line for dumped strings.
+ * @method self withRefHandles(bool $handles)                      Enables/disables objects' global identifiers dumping.
+ * @method self withCasters(array $casters, bool $replace = false) Adds or replace casters for resources and objects.
+ * @method self withFilter(int $filter)                            A bit field of Caster::EXCLUDE_* constants.
+ */
+class Dump
+{
+    const HTML = 'html';
+    const CLI = 'cli';
+    const TEXT = 'text';
+
+    private $vars;
+    private $dumpOnDestruct;
+    private $params = array(
+        'to' => null,
+        'as' => array('env'),
+        'die' => false,
+//        'withTitle' => array(''),
+//        'withTrace' => array(false),
+        'withMaxItems' => array(2500),
+        'withMaxDepth' => array(-1),
+        'withMaxItemsPerDepth' => array(-1),
+        'withMaxStringLength' => array(-1),
+        'withMaxStringWidth' => array(-1),
+        'withRefHandles' => array(true),
+        'withCasters' => array(array(), false),
+        'withFilter' => array(0),
+    );
+
+    public function __construct(array $vars, $dumpOnDestruct = true)
+    {
+        $this->vars = $vars;
+        $this->dumpOnDestruct = $dumpOnDestruct;
+    }
+
+    public function __destruct()
+    {
+        if ($this->dumpOnDestruct) {
+            $this->dumpOnDestruct = false;
+            $this->dump();
+        }
+    }
+
+    public function dump()
+    {
+        extract($this->params);
+
+        if ($withCasters[1]) {
+            $cloner = new VarCloner($withCasters[0]);
+        } else {
+            $cloner = new VarCloner();
+            $cloner->addCasters($withCasters[0]);
+        }
+        $cloner->setMaxItems($withMaxItems[0]);
+        $cloner->setMaxString($withMaxStringLength[0]);
+
+        $dumper = $this->getDumper($this->getFormat($as[0]), $to);
+        $dumper->setMaxStringWidth($withMaxStringWidth[0]);
+
+        foreach ($this->vars as $v) {
+            $dumper->dump($cloner->cloneVar($v, $withFilter[0])
+                ->withMaxDepth($withMaxDepth[0])
+                ->withMaxItemsPerDepth($withMaxItemsPerDepth[0])
+                ->withRefHandles($withRefHandles[0])
+            );
+        }
+
+        if ($die) {
+            exit(1);
+        }
+    }
+
+    protected function getFormat($format)
+    {
+        foreach (array($format, getenv('DUMP_AS')) as $format) {
+            $format = strtolower($format);
+            if ('html' === $format || 'cli' === $format || 'text' === $format) {
+                return $format;
+            }
+        }
+
+        return 'cli' === PHP_SAPI ? 'cli' : 'html';
+    }
+
+    protected function getDumper($format, $output)
```

Comment:
To me, this object is way more a builder than a value object; builders are specialized, and looking at the feedback I see here and here, this one will be specialized... Thus, I'd want it to be possible from the beginning, by inheritance or something better possible.